src/transaction/log_page_buffer.c

```diff
@@ -7359,15 +7362,26 @@ logpb_checkpoint_trantable (THREAD_ENTRY * const thread_p)
 	_er_log_debug (ARG_FILE_LINE, "checkpoint_trantable: failed; writing metalog to file\n");
 	return res_metalog_to_file;
       }
+  }
 
-    logpb_flush_pages (thread_p, &trantable_checkpoint_lsa);
+  // function explicitely needs to be called in critical section-free context
+  logpb_flush_pages (thread_p, &trantable_checkpoint_lsa);
```

Comment:
It would be easier to:
```suggestion
  LOG_CS_EXIT (thread_p);
  logpb_flush_pages (thread_p, &trantable_checkpoint_lsa);
  LOG_CS_ENTER (thread_p);
```
Then making a new scope and repeating scope_exit code would become unnecessary.